File: components/sdk/dotnet/src/Spas.Sdk.Metadata/Builders/NetworkBuilder.cs
using Spas.Sdk.Metadata.Models;

namespace Spas.Sdk.Metadata.Builders;

/// <summary>
/// Builder for network metadata.
/// </summary>
public class NetworkBuilder
{
    private readonly List<string> _requiredEgress = new();

    /// <summary>
    /// Adds a required egress host:port.
    /// </summary>
    public NetworkBuilder AddRequiredEgress(string hostPort)
    {
        _requiredEgress.Add(hostPort);
        return this;
    }

    /// <summary>
    /// Builds the network metadata.
    /// </summary>
    public NetworkMetadata Build()
    {
        return new NetworkMetadata
        {
            RequiredEgress = _requiredEgress
        };
    }
}

File: components/sdk/java/spas-sdk-metadata/src/main/java/io/spas/sdk/metadata/builders/NetworkBuilder.java
package io.spas.sdk.metadata.builders;

import io.spas.sdk.metadata.model.Network;

import java.util.ArrayList;
import java.util.List;

/**
 * Fluent builder for network metadata.
 */
public final class NetworkBuilder {
    private final List<String> requiredEgress = new ArrayList<>();
    
    private NetworkBuilder() {}
    
    public static NetworkBuilder create() {
        return new NetworkBuilder();
    }
    
    public NetworkBuilder addRequiredEgress(String dependency) {
        this.requiredEgress.add(dependency);
        return this;
    }
    
    public Network build() {
        return new Network(requiredEgress.isEmpty() ? null : new ArrayList<>(requiredEgress));
    }
}
